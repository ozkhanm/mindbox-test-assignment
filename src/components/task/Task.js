const Task = ({text, taskStatus, id, taskFinishButtonClickHandler}) => {
    return (
        <div className="task-container">
            <input className="task-status-element" type="checkbox" checked={taskStatus} id={`task${id}`} 
                onChange={() => taskFinishButtonClickHandler(id)} />
            <label className={taskStatus ? "task-text-finished" : ""} htmlFor={`task${id}`}>{text}</label>
        </div>
    );
};

export default Task;