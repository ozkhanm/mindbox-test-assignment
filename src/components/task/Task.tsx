import React from "react";

interface TaskTypes {
    text: string;
    taskStatus: boolean;
    id: number;
    taskFinishButtonClickHandler: (id: number) => void;
};

const Task: React.FC<TaskTypes> = ({text, taskStatus, id, taskFinishButtonClickHandler}) => {
    return (
        <div className="task-container small-container">
            <input className="task-status-element" type="checkbox" checked={taskStatus} id={`task${id}`} 
                onChange={() => taskFinishButtonClickHandler(id)} />
            <label className={taskStatus ? "task-text-finished" : ""} htmlFor={`task${id}`}>{text}</label>
        </div>
    );
};

export default Task;