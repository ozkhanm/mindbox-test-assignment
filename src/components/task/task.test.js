import React from "react";
import renderer from "react-test-renderer";

import Task from "./Task";

it(`Task component renders correctly`, () => {
    const tree = renderer
        .create(<Task />)
        .toJSON();
  
    expect(tree).toMatchSnapshot();
});
