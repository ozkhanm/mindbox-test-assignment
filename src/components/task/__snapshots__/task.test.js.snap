// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component renders correctly 1`] = `
<div
  className="task-container small-container"
>
  <input
    className="task-status-element"
    id="taskundefined"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className=""
    htmlFor="taskundefined"
  />
</div>
`;
